endpoints/openrtb2/amp_auction.go

```diff
@@ -219,6 +220,7 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 		ao.Errors = append(ao.Errors, err)
 		return
 	}
+	labels.DebugEnabled = debugLog.DebugFlag
```

Comment:
I consider it a "code smell" that `debugLog` is instantiated here just to return the DebugFlag to the outer scope. It would be preferable IMHO to either calculate the debug flag prior to calling `HoldAuction` or include it as a return argument somehow. I know you didn't create this situation, but I believe it's worth an investigation to try and make this flow a bit nicer now.

Reply:
Would it be better if labels.DebugEnabled will be passed in `auctionRequest` downstrem to `getDebugInfo` and modified there? 
I cannot think about better solution...